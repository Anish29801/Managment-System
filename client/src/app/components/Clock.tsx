// Clock.tsx
"use client";
import { useEffect, useState } from "react";

export default function Clock() {
  const [time, setTime] = useState(getCurrentTime());

  function getCurrentTime() {
    const now = new Date();
    let hours = now.getHours();
    const minutes = now.getMinutes();
    const ampm = hours >= 12 ? "PM" : "AM";
    hours = hours % 12 || 12;
    const minutesStr = minutes < 10 ? `0${minutes}` : minutes;
    return `${hours}:${minutesStr} ${ampm}`;
  }

  useEffect(() => {
    const timer = setInterval(() => {
      setTime(getCurrentTime());
    }, 1000);
    return () => clearInterval(timer);
  }, []);

  return (
    <div className="text-white font-semibold text-base sm:text-lg md:text-xl lg:text-2xl">
      {time}
    </div>
  );
}
