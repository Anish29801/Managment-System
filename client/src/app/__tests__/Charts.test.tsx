import { render, screen } from "@testing-library/react";
import React from "react";
import Clock from "@/app/components/Clock";

// ğŸ§© Mock the Clock component for isolated testing
jest.mock("@/app/components/Clock", () => ({
  __esModule: true,
  default: () => <div data-testid="clock">Mock Clock</div>,
}));

describe("Clock component", () => {
  // âœ… 1. Renders mock Clock component
  it("renders Clock", () => {
    render(<Clock />);
    expect(screen.getByText("Mock Clock")).toBeInTheDocument();
  });

  // âœ… 2. Has specific test id
  it("has data-testid attribute", () => {
    render(<Clock />);
    expect(screen.getByTestId("clock")).toBeInTheDocument();
  });

  // âœ… 3. Ensures only one instance renders
  it("renders only one instance", () => {
    render(<Clock />);
    const elements = screen.getAllByText("Mock Clock");
    expect(elements).toHaveLength(1);
  });

  // âœ… 4. Rerenders cleanly when props change (no crash)
  it("renders again when re-rendered", () => {
    const { rerender } = render(<Clock />);
    rerender(<Clock />);
    expect(screen.getByText("Mock Clock")).toBeInTheDocument();
  });

  // âœ… 5. Snapshot test for stable output
  it("matches snapshot", () => {
    const { asFragment } = render(<Clock />);
    expect(asFragment()).toMatchSnapshot();
  });
});
