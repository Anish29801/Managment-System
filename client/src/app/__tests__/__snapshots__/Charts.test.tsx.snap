// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Clock component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="clock"
  >
    Mock Clock
  </div>
</DocumentFragment>
`;
